cores/rp2040/delay.cpp

```diff
@@ -21,30 +21,42 @@
 #include <pico.h>
 #include <pico/time.h>
 
-extern "C" void delay( unsigned long ms ) {
+#ifdef USE_TINYUSB
+#include "Adafruit_TinyUSB_API.h"
+#endif
+
+extern "C"
```

Comment:
The rp2040 port from the Arduino Lib is changed to call the tud_task() from irq context like "pico-sdk" stack. The TinyUSB_Device_Task() is also implemented in the way the require mutex to run. Therefore it no longer needs to use the task ready event API as previous commit. 

The reason for not pursuing thread-context task, is that the sleep_ms() from pico-sdk call wfe() does not provide hook() to run background work in thread-context. In short CPU will wake in case of USB IRQ, run the handler and then sleep again without any chances to run tud_task() in thread-context.

PS: Let me know if you want to revert to have extern "C" for each function instead of block {}.